models/DataObject/ClassDefinition/Data/Block.php

```diff
@@ -127,6 +127,24 @@ public function getDataForResource($data, $object = null, $params = [])
 
                 /** @var DataObject\Data\BlockElement $blockElement */
                 foreach ($blockElements as $elementName => $blockElement) {
+
+                    if (!isset($params['owner'])) {
```

Comment:
Should we maybe move this code to a `private` function to avoid duplicate code?

Reply:
done